packages/zowe-explorer-api/src/logger/IZoweLogger.ts

```diff
@@ -13,6 +13,23 @@ import * as loggerConfig from "../log4jsconfig.json";
 import * as path from "path";
 import { imperative } from "@zowe/cli";
 
+type Appender = {
+    type: string;
+    layout: {
+        type: string;
+        pattern: string;
+    };
+    filename: string;
+};
+
+type Log4JsCfg = {
+    log4jsConfig: {
+        appenders: { [key: string]: Appender };
+    };
+};
```

Comment:
Thanks for pointing those out - I think we might be able to use those. I noticed that since the `ILog4jsAppender` object is indexable, when I change to those types it brings back the "no unsafe argument" error as the typing for the values are defined as `any` for the appender:

```js
export interface ILog4jsAppender {
    "type": string;
    "layout": ILog4jsLayout;
    [key: string]: any;
}
```

If we know the types for the appender values, we could always define those within that imperative interface and that should indirectly fix the error.

Reply:
We could also extend the Imperative interface for our usage here:
```
export interface ILog4jsAppender extends imperative.ILog4jsAppender {
    [key: string]: Appender;
}
```

No change required, I'm cool with leaving it as-is 🙂